### PR TITLE
SMT2 solver: implement get-value

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [X] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: d6f7c5a3ebddd24d67f9af3345c0690e0b79ef45).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/92464624

File: src/solvers/smt2/smt2_solver.cpp [hunk 1]
```diff
@@ -39,6 +38,13 @@ class smt2_solvert:public smt2_parsert
   void expand_function_applications(exprt &);
 
   std::set<irep_idt> constants_done;
+
+  enum
+  {
+    NOT_SOLVED,
+    SAT,
+    UNSAT
+  } status;
```
Comment (tautschnig) [inline src/solvers/smt2/smt2_solver.cpp:47]: Why do we need yet another status enum? We have one for decision procedures, one in prop.h - wouldn't the latter be the right one?

**Review by tautschnig — APPROVED** [review]

File: src/solvers/smt2/smt2_solver.cpp (hunk 1 above)
Comment (kroening) [reply src/solvers/smt2/smt2_solver.cpp:47]: No; this isn't a result from a solver, but the state of the SMT2-LIB protocol between the solver and the client. The key question is whether (check-sat) has been called or not.

Comment (tautschnig) [reply src/solvers/smt2/smt2_solver.cpp:47]: Maybe such things could be documented in future? The code only did a straightforward translation from enum values of one type to enum values of another type.

**Pull request merged**